xonsh/proc.py

```diff
@@ -12,46 +12,49 @@
 import sys
 import time
 import builtins
-from functools import wraps
-from threading import Thread
-from collections import Sequence, namedtuple
-from subprocess import (Popen, PIPE, DEVNULL, STDOUT, TimeoutExpired,
-                        CalledProcessError)
+import functools
+import threading
+import importlib
+import subprocess
+import collections
+import collections.abc as abc
 
 from xonsh.tools import (redirect_stdout, redirect_stderr, ON_WINDOWS, ON_LINUX,
                          fallback, print_exception, XonshCalledProcessError)
+from xonsh.teepty import TeePTY
```

Comment:
Do you know which ones?  It seems like the right fix is to make those imports explicitly lazy / non-windows.